### PR TITLE
Cleaned up Element range handling

### PR DESCRIPTION
**Pull request by philippjfr**
As described in https://github.com/ioam/holoviews/issues/1870, QuadMesh is currently not handling explicitly set dimension ranges. This PR adds a utility to combine the data range with the dimension ranges and uses it in various Element.range implementations.

### CODE REVIEW

File: holoviews/core/util.py [hunk 1]
```diff
@@ -710,6 +710,18 @@ def max_range(ranges):
         return (np.NaN, np.NaN)
 
 
+def dimension_range(lower, upper, dimension):
+    """
+    Computes the range along a dimension by combining the data range
+    with the Dimension soft_range and range.
+    """
```
Comment (jlstevens) [inline holoviews/core/util.py]: I would be surprised if we don't have very similar logic to this somewhere else in holoviews already...

Comment (jlstevens) [reply holoviews/core/util.py]: I see you replaced two chunks of code with this utility below, is that everywhere this utility could be used?

**Comment by jlstevens** [comment]
Looks good. 

It would be good if you could check that ``dimension_range`` can't be used in more places (I suspect so, but I might be wrong!). Otherwise happy to merge.

**Comment by philippjfr** [comment]
I think this was long overdue tbh, there were various inconsistencies in the implementations.

**Comment by jlstevens** [comment]
Tests are green, if you think you've used this new utility everywhere that makes sense, I'm happy to merge.

**Comment by philippjfr** [comment]
Yes, that's every single range implementation now.

**Pull request merged**